swap_meet/vendor.py

```diff
@@ -75,33 +78,34 @@ def swap_best_by_category(self, other_vendor, my_priority, their_priority):
         my_best_item = self.get_best_by_category(their_priority)
         their_best_item = other_vendor.get_best_by_category(my_priority)
 
-        if my_best_item is None or their_best_item is None:
-            return False
-
         return self.swap_items(other_vendor, my_best_item, their_best_item)
-    
+        
 # ================Optional Enhancements================
 
-    def _get_newest_item_by_condition(self):
+    def get_newest_item_by_condition(self):
         """
         Return the newest item based on age, using condition as a tiebreaker.
+        Items without an age value are filtered out before comparison.
         """
 
-        best_item = self.inventory[0]
-        newest_age = best_item.age
-        best_condition = best_item.condition
+        valid_items = [item for item in self.inventory if item.age is not None]
```

Comment:
👍 Nice use of a list comprehension to filter out the results to only those that have a known age.